synapse/federation/transaction_queue.py

```diff
@@ -207,6 +213,19 @@ def _process_event_queue_loop(self):
 
                     self._send_pdu(event, destinations)
 
+                def handle_room_events(events):
+                    for event in events:
+                        return handle_event(event)
```

Comment:
That will handle the first event of events and skip the other events of the queue - even if that is valid or not

Reply:
Good catch